tools/serverless-plugin/src/generators/service/notification-files/src/utils/aws-formatting.ts.template

```diff
@@ -1,103 +1,217 @@
+ * const name = extractNameFromExecutionArn(arn);
+ * console.log(name); // Output: 'my-function'
  */
 export function extractNameFromExecutionArn(arn: string) {
     const parts = arn.split(':');
-    const name = parts[6];
+    let name = parts[6];
 
     if (!name) {
         throw new Error('Invalid execution ARN');
     }
 
+    // If it's a Step Function execution ARN, we want to include execution name
+    if (parts.length === 8) {
+        name += `: ${parts[7]}`;
+    }
+
     return name;
 }
 
 /**
  * Extracts the name from a resource name string.
  * It assumes our resource name is in the correct format
- * Eg: brand-int-service-name-stage-resourceName
+ * For example:
+ * - Lambda resource name format is: brand-int-service-name-stage-lambdaName
+ * - Step function resource name format is: brand-int-service-name-stage-stepFunctionName:994ee932-8e5f-f59f-5619-ebdefd8af815_087eb4a0-92a4-30d2-eea4-c078e9b9e5fa
  *
  * @param {string} resourceName - The resource name string.
  * @returns {string} The extracted name from the resource name.
+ *
+ * @example
+ * const resourceName = 'brand-int-service-name-stage-lambdaName';
+ * const name = extractNameFromResourceName(resourceName);
+ * console.log(name); // Output: 'lambdaName'
+ *
+ * @example
+ * const resourceName = 'brand-int-service-name-stage-stepFunctionName:994ee932-8e5f-f59f-5619-ebdefd8af815_087eb4a0-92a4-30d2-eea4-c078e9b9e5fa';
+ * const name = extractNameFromResourceName(resourceName);
+ * console.log(name); // Output: 'stepFunctionName'
  */
 export function extractNameFromResourceName(resourceName: string) {
+    // Only Step Function resource name contains : character.
+    if (resourceName.includes(':')) {
+        const [stateMachineName, executionName] = resourceName.split(':');
+
+        // This is a scheduled execution, we will use the state machine name instead
+        resourceName = (executionName?.includes('_') ? stateMachineName : executionName) as string;
+    }
+
     return resourceName.split('-').pop() as string;
 }
 
 /**
  * Extracts the error details from the error detail object
+ * Notes:
+ *  - Lambda sends a JSON string in detail.cause
+ *  - Step Function Fail step sends an object that contains both detail.cause & detail.error
+ *  - When Step Function rethrow lambda error, it sends detail.cause is not a JSON string
+ *
  * @param {NotifyError.InvocationEvent['detail']} detail - The detail object to extract the error details from
  * @returns {NotifyError.Cause} The error details
+ *
+ * @example
+ * const detail = {
+ *   cause: '{"errorType":"MyError","errorMessage":"Something went wrong"}',
+ *   error: "MyError"
+ * };
+ * const errorDetails = extractErrorDetails(detail);
+ * console.log(errorDetails);
+ * // Output: { errorType: "MyError", errorMessage: "Something went wrong" }
  */
 export function extractErrorDetails(
-    event: NotifyError.InvocationEvent
+    detail: NotifyError.InvocationEvent['detail']
```

Comment:
@toddhainsworth I have to rewrite this function because it has type errors. I also added some explaination in the JSDoc on why I did it this way.

Reply:
I don't get a type error if I write it like this

```typescript
export function extractErrorDetails(event: NotifyError.InvocationEvent): NotifyError.Cause {
    const { cause, error } = event.detail;
```

What error were you getting? Would prefer not to change the interface here.

Reply:
I change the inferface because the original JSDoc say that the param should be `NotifyError.InvocationEvent['detail']`. 

Regarding the type error, I get it when returning `NotifyError.Cause` object as it requires both `errorType` and `errorMessage` to be strings.
![image](https://github.com/user-attachments/assets/c15dd07a-a270-41cf-b68e-c909192051a7)